### ISSUE TITLE
libxml2 not found while running ./configure 

### ISSUE THREAD

**Issue by IanGruson** [pgpointcloud/pointcloud#263]
Hello, 
I'm encountering a problem while using autotools to build this plugin. 
I have libxml2 2.9.10 installed and located in /usr/include/libxml2

but when I run ./configure I get the following error : 
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking for cpp... /usr/bin/cpp
checking for git... yes
checking for .git... yes
checking for g++... g++
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking how to run the C++ preprocessor... g++ -E
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking zlib.h usability... yes
checking zlib.h presence... yes
checking for zlib.h... yes
checking for inflate in -lz... yes
checking CUnit/CUnit.h usability... yes
checking CUnit/CUnit.h presence... yes
checking for CUnit/CUnit.h... yes
checking for CU_initialize_registry in -lcunit... yes
checking for pg_config... /usr/bin/pg_config
checking PostgreSQL version... PostgreSQL 11.7 (Ubuntu 11.7-2.pgdg18.04+1)
checking libpq-fe.h usability... yes
checking libpq-fe.h presence... yes
checking for libpq-fe.h... yes
checking for PQserverVersion in -lpq... yes
checking for xml2-config... /usr/bin/xml2-config
checking libxml/tree.h usability... yes
checking libxml/tree.h presence... yes
checking for libxml/tree.h... yes
checking libxml/parser.h usability... yes
checking libxml/parser.h presence... yes
checking for libxml/parser.h... yes
checking libxml/xpath.h usability... yes
checking libxml/xpath.h presence... yes
checking for libxml/xpath.h... yes
checking libxml/xpathInternals.h usability... yes
checking libxml/xpathInternals.h presence... yes
checking for libxml/xpathInternals.h... yes
checking for xmlInitParser in -lxml2... no
configure: error: could not find libxml2

Thanks in advance

**Comment by pblottiere** [comment]
Hello @IanGruson,

What is your distribution please? I'll try to reproduce the error locally.

**Comment by IanGruson** [comment]
hello @pblottiere,

I'm using Ubuntu 18.04.4

**Comment by pblottiere** [comment]
> I'm using Ubuntu 18.04.4

Thanks, I'll take a look asap :+1:

**Comment by pblottiere** [comment]
@IanGruson,

I just tried on a fresh Docker image with Ubuntu 18.04.4 and installing `libxml2-dev` seems to be enough for the configure check:

``` bash
checking for xml2-config... /usr/bin/xml2-config
checking libxml/tree.h usability... yes
checking libxml/tree.h presence... yes
checking for libxml/tree.h... yes
checking libxml/parser.h usability... yes
checking libxml/parser.h presence... yes
checking for libxml/parser.h... yes
checking libxml/xpath.h usability... yes
checking libxml/xpath.h presence... yes
checking for libxml/xpath.h... yes
checking libxml/xpathInternals.h usability... yes
checking libxml/xpathInternals.h presence... yes
checking for libxml/xpathInternals.h... yes
checking for xmlInitParser in -lxml2... yes
...
```

Can you give me the output for the next commands please?

``` bash
$ xml2-config --prefix
/usr
$ xml2-config --cflags
-I/usr/include/libxml2
$ xml2-config --version
2.9.4
$ xml2-config --libs
-lxml2
```

**Comment by IanGruson** [comment]
@pblottiere 

Thank you for your time, 

I do have libxml2-dev 2.9.10 installed aswell using apt.

here are the outputs for the commands 

$ xml2-config --prefix
/usr
$ xml2-config --cflags
-I/usr/include/libxml2
$ xml2-config --version
2.9.10
$ xml2-config --libs
-lxml2 -lz -llzma -licui18n -licuuc -licudata -lm -ldl
[ian@ian-ux430unr ~]$ 

It appears we aren't using the same version, could it be the issue ?

**Comment by IanGruson** [comment]
Hello again, It appears the version 2.9.10 doesn't work, but 2.9.4 of libxml does. I didn't know. My apt would only install 2.9.10, I had to install by hand. It took me a while before trying this solution.

**Comment by pblottiere** [comment]
Thanks for the feedback @IanGruson :+1: 

I keep this issue opened because I'd like to investigate a little more with libxml 2.9.10.

**Comment by IanGruson** [comment]
No problem, 

Let me now if 2.9.10 works on your environment and if it was just something I did wrong.

**Comment by pblottiere** [comment]
BTW, how did you install libxml 2.9.10 @IanGruson? Because on Ubuntu 18.04, the default version is 2.9.4 (cf https://launchpad.net/ubuntu/bionic/+source/libxml2).

**Comment by pblottiere** [comment]
> Let me now if 2.9.10 works on your environment and if it was just something I did wrong.

I just tested with libxml 2.9.10 without issue:

```
$ xml2-config --version
2.9.10
$ xml2-config --libs
-lxml2 -lz -llzma -licui18n -licuuc -licudata -lm -ldl
$ xml2-config --prefix
/usr
$ ./configure
...
checking for xml2-config... /usr/bin/xml2-config
checking libxml/tree.h usability... yes
checking libxml/tree.h presence... yes
checking for libxml/tree.h... yes
checking libxml/parser.h usability... yes
checking libxml/parser.h presence... yes
checking for libxml/parser.h... yes
checking libxml/xpath.h usability... yes
checking libxml/xpath.h presence... yes
checking for libxml/xpath.h... yes
checking libxml/xpathInternals.h usability... yes
checking libxml/xpathInternals.h presence... yes
checking for libxml/xpathInternals.h... yes
checking for xmlInitParser in -lxml2... yes
checking las.hpp usability... no
checking las.hpp presence... no
checking for las.hpp... no
configure: creating ./config.status
config.status: creating config.mk
config.status: creating pgsql/Makefile
config.status: creating lib/pc_config.h
config.status: creating pgsql/sqldefines.h
config.status: pgsql/sqldefines.h is unchanged

  PointCloud is now configured for

 -------------- Compiler Info -------------
  C compiler:           gcc -g -O2
  SQL preprocessor:     /usr/bin/cpp -traditional-cpp -w -P

 -------------- Dependencies --------------
  PostgreSQL config:    /usr/bin/pg_config
  PostgreSQL version:   PostgreSQL 12.2 (122)
  Libxml2 config:       /usr/bin/xml2-config
  Libxml2 version:      2.9.10
  LazPerf status:       disabled
  CUnit status:         enabled
```

**Comment by IanGruson** [comment]
> BTW, how did you install libxml 2.9.10 @IanGruson? Because on Ubuntu 18.04, the default version is 2.9.4 (cf https://launchpad.net/ubuntu/bionic/+source/libxml2).

I use sudo apt install libxml2 && sudo apt install libxml2-dev which got the 2.9.10 version.
When I tried to downgrade versions with sudo apt install libxml2=2.9.4 it wouldn't work actually. I had to download the libxml2-2.9.4 archive and build it manually. Something with my sources.list file ? 

Sorry if I get things all wrong.

**Comment by pblottiere** [comment]
> I use sudo apt install libxml2 && sudo apt install libxml2-dev which got the 2.9.10 version.
When I tried to downgrade versions with sudo apt install libxml2=2.9.4 it wouldn't work actually. I had to download the libxml2-2.9.4 archive and build it manually. Something with my sources.list file ?

When I try to install libxml2-dev with apt, 2.9.4 is still the newest version available on Ubuntu 18.04.4:

```
# lsb_release -a
Distributor ID:Ubuntu
Description:	Ubuntu 18.04.4 LTS
Release:	18.04
Codename:	bionic
# apt install libxml2-dev
Reading package lists... Done
Building dependency tree
Reading state information... Done
libxml2-dev is already the newest version (2.9.4+dfsg1-6.1ubuntu1.3).
0 upgraded, 0 newly installed, 0 to remove and 9 not upgraded.
```

> Something with my sources.list file ?

Yes maybe. Can you give your source.list please?

**Comment by pblottiere** [comment]
No response after 3 weeks, I'm closing the issue.
